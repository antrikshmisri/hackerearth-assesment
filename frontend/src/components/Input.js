import React from "react";
import { Container, Row, Col } from "react-bootstrap";

const Input = ({
  icon,
  value,
  placeholder,
  onChange,
  onPaste,
  required,
  label,
  keyPressCallback = () => {},
}) => {
  return (
    <Container className="px-0 py-0">
      <Row>
        <Col lg={12} className="d-flex justify-content-center">
          <p style={{ fontWeight: "500" }}>{label}</p>
          {icon ? <span className="input-icon-div">{icon}</span> : null}
          <input
            type="text"
            value={value}
            placeholder={placeholder}
            onChange={(e) => {
              onChange(e);
              keyPressCallback(e);
            }}
            required={required}
          ></input>
        </Col>
      </Row>
    </Container>
  );
};

export default Input;
