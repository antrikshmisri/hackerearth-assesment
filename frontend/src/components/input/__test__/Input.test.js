import React from "react";
import ReactDOM from "react-dom";

import Input from "../Input";
import { AiFillAlert } from "react-icons/ai";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Input
      icon={<AiFillAlert />}
      placeholder="Test Input Field"
      onChange={() => {}}
      label="Test Label"
      required
    />,
    div
  );
});

it("matches snapshot", () => {
  const inputTree = renderer
    .create(
      <Input
        icon={<AiFillAlert />}
        placeholder="Test Input Field"
        onChange={() => {}}
        label="Test Label"
        required
      />
    )
    .toJSON();

  expect(inputTree).toMatchSnapshot();
});
