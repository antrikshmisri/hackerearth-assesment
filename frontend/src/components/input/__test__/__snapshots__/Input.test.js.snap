// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="px-0 py-0 container"
>
  <div
    className="row"
  >
    <div
      className="d-flex justify-content-center col-lg-12"
    >
      <p
        style={
          Object {
            "fontWeight": "500",
          }
        }
      >
        Test Label
      </p>
      <span
        className="input-icon-div"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 244c176.18 0 319 142.82 319 319v233a32 32 0 0 1-32 32H225a32 32 0 0 1-32-32V563c0-176.18 142.82-319 319-319zM484 68h56a8 8 0 0 1 8 8v96a8 8 0 0 1-8 8h-56a8 8 0 0 1-8-8V76a8 8 0 0 1 8-8zM177.25 191.66a8 8 0 0 1 11.32 0l67.88 67.88a8 8 0 0 1 0 11.31l-39.6 39.6a8 8 0 0 1-11.31 0l-67.88-67.88a8 8 0 0 1 0-11.31l39.6-39.6zm669.6 0l39.6 39.6a8 8 0 0 1 0 11.3l-67.88 67.9a8 8 0 0 1-11.32 0l-39.6-39.6a8 8 0 0 1 0-11.32l67.89-67.88a8 8 0 0 1 11.31 0zM192 892h640a32 32 0 0 1 32 32v24a8 8 0 0 1-8 8H168a8 8 0 0 1-8-8v-24a32 32 0 0 1 32-32zm148-317v253h64V575h-64z"
          />
        </svg>
      </span>
      <input
        onChange={[Function]}
        placeholder="Test Input Field"
        required={true}
        type="text"
      />
    </div>
  </div>
</div>
`;
